File: 482.license-key-formatting.js
/*
 * @lc app=leetcode id=482 lang=javascript
 *
 * [482] License Key Formatting
 */

// @lc code=start
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var licenseKeyFormatting = function(s, k) {
    let copy = s.split('-')
    let res = []
    for (let i = copy.length - k; i >= 0; i -= k) {
        res.unshift(copy.splice(i, copy.length, k).toUpperCase())
    }
    return res.join('-')
};
// @lc code=end

File: 482.license-key-formatting.py
#
# @lc app=leetcode id=482 lang=python
#
# [482] License Key Formatting
#

# @lc code=start
class Solution(object):
    def licenseKeyFormatting(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """

        # Approach 1: String
        # Time  complexity: O(n)
        # Space complexity: O(n)
        s = s.replace("-", "").upper()
        res = ""
        for i in range(len(s) - 1, -1, -1):
            res = s[i] + res
            if i > 0 and (len(s) - i) % k == 0:
                res = "-" + res
        return res

# @lc code=end
